functions.sh

```diff
@@ -321,11 +321,20 @@ curl-md5sum(){
 }
 
 epoch(){
-  if [[ ${1:-UNSET} == "UNSET" ]]; then
-    date '+%s'
-  else
-    /usr/bin/ruby -rdate -e "puts Date.parse('${1}').strftime('%s')"
+  if [[ -z "${1}" ]]; then
+    if type -P date &> /dev/null; then
+      date '+%s'
+      return $?
+    fi
+  fi
+
+  if type -p ruby &> /dev/null; then
+    ruby -rdate -e "puts Date.parse('${1}').strftime('%s')"
+    return $?
   fi
+
+  error "unable to print epoch (no \`date\` or \`ruby\` found in PATH)"
+  return 1
```

Comment:
![giphy](https://user-images.githubusercontent.com/47807687/53392979-eb09d300-394f-11e9-84e9-70f6fda491cf.gif)

I'm _fudging_ tired of the fact that you don't fix problems in the code **you** write, so that we then have to work around the problems you cause. I will *not* be merging any code from you into here until this constant pattern is fixed.